server/src/test/java/org/elasticsearch/index/mapper/IgnoredSourceFieldMapperTests.java

```diff
@@ -759,6 +759,49 @@ public void testArrayWithinArray() throws IOException {
             {"path":[{"to":[{"name":"A"},{"name":"B"}]},{"to":[{"name":"C"},{"name":"D"}]}]}""", booleanValue), syntheticSource);
     }
 
+    public void testObjectArrayAndValue() throws IOException {
+        DocumentMapper documentMapper = createMapperService(syntheticSourceMapping(b -> {
+            b.startObject("path");
+            {
+                b.field("type", "object");
+                b.startObject("properties");
+                {
+                    b.startObject("stored");
+                    {
+                        b.field("type", "object").field("store_array_source", true);
+                        b.startObject("properties").startObject("leaf").field("type", "integer").endObject().endObject();
+                    }
+                    b.endObject();
+                }
+                b.endObject();
+            }
+            b.endObject();
+        })).documentMapper();
+        // { "path": [ { "stored":[ { "leaf": 10 } ] }, { "stored": { "leaf": 20 } } ] }
+        var syntheticSource = syntheticSource(documentMapper, b -> {
+            b.startArray("path");
+            {
+                b.startObject();
+                {
+                    b.startArray("stored");
+                    {
+                        b.startObject().field("leaf", 10).endObject();
+                    }
+                    b.endArray();
+                }
+                b.endObject();
+                b.startObject();
+                {
+                    b.startObject("stored").field("leaf", 20).endObject();
+                }
+                b.endObject();
+            }
+            b.endArray();
+        });
+        assertEquals("""
+            {"path":[{"stored":[{"leaf":10}]},{"stored":{"leaf":20}}]}""", syntheticSource);
```

Comment:
Correct, but then the added test fails.

Reply:
I see what you're saying.. yeah that would be acceptable, but I'm not sure if it would work.. Will test it.

Reply:
So if we don't store the parent array, we get `{"path":{"stored":[{"leaf":10}]}}`

What happens is that the second parsing finds the first match for `stored`, containing `{leaf: 10}`. `stored` is  removed from the map with fields to check for, so it doesn't add a second one - and if it did, it would overwrite the first. one.